src/openms/source/FORMAT/DATAACCESS/CsiFingerIdMzTabWriter.cpp

```diff
@@ -64,10 +64,13 @@ void CsiFingerIdMzTabWriter::read(const std::vector<String> & sirius_output_path
     {
       CsvFile compounds(pathtocsicsv, '\t');
       const UInt rowcount = compounds.rowCount();
-    
+
       if (rowcount > 1)
       {
-        const UInt top_n_hits_cor = (top_n_hits >= rowcount) ? rowcount : top_n_hits;
+        // correction if the rowcount is smaller than the number of hits used as parameter
```

Comment:
Minor suggestion as I found the comments/variable names could be a bit more descriptive.
you could define
``` 
const UInt n_header_rows = 1;
const UInt n_hits_rows = rowcount - n_header_rows;
and
const UInt n_hits_to_extract = std::min(top_n_hits, n_hits_rows);
```
instead of using the top_n_hits_cor variable name .
comment in line 92 would probably than not be necessary
and you could loop from j=0 to n_hits_to_extract but inside of the loop call .getRow(n_header_rows + j) instead.
All in all slightly clearer (at least to me) :P